### ISSUE TITLE
Support for other json derived content types

### ISSUE THREAD

**Issue by andrealexandre** [guardrail-dev/guardrail#1119]
I noticed there's no way to configure or allow the client/server generator to allow to support more than just the traditional `application/json` encoder.
Just on the Akka MediaTypes singleton there are these 3, e.g.:
* `application/json-patch+json`
* `application/merge-patch+json`
* `application/problem+json`

I'm more than happy to open a PR, but wanted to see if this is something in the scope of this library.

Ref:
https://github.com/guardrail-dev/guardrail/blob/9646130eaa7c344b58a93a5f8cdf7ac6a883a0ac/modules/codegen/src/main/scala/com/twilio/guardrail/generators/Scala/AkkaHttpGenerator.scala#L195

**Comment by kelnos** [comment]
I think that'd be a reasonable thing to support (I assume you actually have a use case that needs this?).  I think what we'd need to do is push this up a little higher into the `ContentType` enum, but change that to a case class so it can store the actual content-type desired, not just generic `application/json`.  That way it'd make it easier to add support to all the generators, not just akka-http.

**Comment by andrealexandre** [comment]
@kelnos yes, I have a use case for `application/problem+json`, the ideal would be to just ref `application/*+json`.
But yes, allowing to set custom content types would allow for interesting stuff, like using yaml or xml.

**Comment by andrealexandre** [comment]
Either way, I opened a PR to hopefully jump-start some progress on this, given that Akka supports these content types, I hope it's not much of a problem to add it as hardcoded to guardrail for now.
https://github.com/guardrail-dev/guardrail/pull/1120

**Comment by andrealexandre** [comment]
Just a brief update in case someone is struggling with this issue, a quick and dirty hack is to add some logic to the httpClient function the client has as a dependency. `HttpRequest => Future[HttpResponse] `

```
private def ensureContentType(response: HttpResponse) = {
    response.headers.map {
      case contentType: ContentType =>
        if (contentType.contentType().mediaType == MediaTypes.`application/problem+json`) {
          ContentType.create(WithFixedCharset(MediaTypes.`application/json`))
        } else {
          contentType
        }
      case header => header
    }
  }

implicit val httpClient: HttpRequest => Future[HttpResponse] = { request =>
  httpExt.singleRequest(request).map { response =>
     response.withHeaders(ensureContentType(response))
  }
}
```
